src/lib/tools/currency_converter.py

```
@@ -24,62 +24,72 @@ def __init__(self):
-                    elif line == "amount":
-                        raise ConversionRequestError(
-                            "Missing required parameter: amount"
-                        )
-                raise ConversionRequestError("Missing required parameter")
-            elif "Input should be greater than 0" in error_msg:
-                raise ConversionRequestError("Amount must be a positive number")
-            elif "Input should be a valid number" in error_msg:
-                raise ConversionRequestError("Amount must be a positive number")
-            else:
-                raise ConversionRequestError(f"Invalid conversion request: {error_msg}")
+            raise self._handle_value_error(e)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)